SingularityExecutor/src/main/java/com/hubspot/singularity/executor/SingularityExecutorMonitor.java

```diff
@@ -83,6 +102,10 @@ public SingularityExecutorMonitor(@Named(SingularityExecutorModule.ALREADY_SHUT_
     this.threadChecker = threadChecker;
     this.threadChecker.start(this);
 
+    this.requestManager = requestManager;
+    this.deployManager = deployManager;
+    this.taskManager = taskManager;
```

Comment:
SingularityData stuff would be a very heavy dep to add into every executor. I do not think we want to add these here.
- `requestManager` data your fetching here isn't actually being used
- `HealthcheckOptions` we should feed in through the `SingularityTaskExecutorData` like you are doing with the path
- There is no valuable information in saving health check results for the task since all we are doing is checking a file. There will essentially be a single pass/fail health check result, and we can even leave that as encoded in the task status update to keep things simpler if we want. The basics would be that the status update of running serves as our healthcheck success, or a status update of killing/failed/etc serves as our failure.